### ISSUE TITLE
Out-of-memory on g2.8xlarge

### ISSUE THREAD

**Issue by lukeyeager** [NVIDIA/caffe#34]
See https://github.com/NVIDIA/DIGITS/issues/310.

/cc @ajsander

> I've trained a couple models (Alexnet and GoogleNet) using DIGITS successfully with statistics shown for test and validation accuracy, but when I try to classify a single image using the web interface I get the following error:

>    WARNING: Logging before InitGoogleLogging() is written to STDERR
    F0915 14:10:45.809661 98789 common.cpp:266] Check failed: error == cudaSuccess (2 vs. 0)  out of        memory
    *** Check failure stack trace: ***

>When I check nvidia-smi it appears that it the amount of memory is increasing by around 100MB but it's still nowhere near the full memory capacity of the card at 3GB. 
https://github.com/NVIDIA/DIGITS/issues/310#issue-106567556

Here is some information about his system:

> Running on an Amazon g2.8xlarge
GPU[s]: 4x GRID K520
CUDA 7.0
cuDNN 7.0
Caffe version 0.12 NVIDIA fork
DIGITS 2.1

>Both Alexnet and GoogleNet Experienced the same problem
https://github.com/NVIDIA/DIGITS/issues/310#issuecomment-140457927

Here's how I reproduced it:

1. Start up a Ubuntu 14.04 on `g2.8xlarge` EC2 instance
2. Install the 346 driver
3. Installed DIGITS 2.0 and Caffe 0.13.1 (with CNMeM) using the [web installer](https://developer.nvidia.com/digits)
4. Create a small dataset of 256x256 images
5. Train AlexNet on it
6. Try to classify an image

## The big question
Why would we run out of memory during inference but not while training?

**Comment by creitiv** [comment]
I also get the same issue when trying to classify an image at very low memory load.

My card is a GTX 660 Ti with 2 GB, but the memory usage when running into the error is only about 10%.

System is Ubuntu 15.04 and i'm using the most recent version of DIGITS and nvidia's caffe fork.

After getting the error, the model also disappears completely. It doesn't matter if it's trained to the end or not, it's just disappears from the model list, and i get 404 errors when trying to open it.

A log that appears all the time is also
```
Caught PicklingError while saving job: Can't pickle <class 'caffe_pb2.NetParameter'>: it's not found as caffe_pb2.NetParameter
```
but i don't now if it's related with the bug somehow.

**Comment by drozdvadym** [comment]
+1, get the same error

* GPU Usage GeForce GTX 660 Ti (#0)
* Linux 3.13.0-63-generic #103-Ubuntu SMP Fri Aug 14 21:42:59 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
* caffe NVIDIA fork 0.13, DIGITS 2.2.1

**Comment by lukeyeager** [comment]
See here also:
https://groups.google.com/d/msg/digits-users/8-Bqik4nECI/HDQmqzSuBQAJ

Oddly, his problem went away by upgrading to the latest Caffe and DIGITS.

**Comment by lukeyeager** [comment]
Seems to be solved for this guy with v0.14:
http://www.learnopencv.com/nvidia-digits-3-on-ec2/
